slothy/targets/arm_v7m/arch_v7m.py

```diff
@@ -0,0 +1,1882 @@
+        yield f"cmp {cnt}, #{val}"
+        yield f"beq {lbl}"
+
+    @staticmethod
+    def if_greater_equal(cnt, val, lbl):
+        """Emit assembly for a branch-if-greater-equal sequence"""
+        yield f"cmp {cnt}, #{val}"
+        yield f"bge {lbl}"
+
+    @staticmethod
+    def unconditional(lbl):
+        """Emit unconditional branch"""
+        yield f"b {lbl}"
+
+
+class VmovCmpLoop(Loop):
+    def __init__(self, lbl="lbl", lbl_start="1", lbl_end="2", loop_init="lr") -> None:
+        super().__init__(lbl_start=lbl_start, lbl_end=lbl_end, loop_init=loop_init)
+        self.lbl = lbl
+        self.lbl_regex = r"^\s*(?P<label>\w+)\s*:(?P<remainder>.*)$"
+        self.end_regex = (r"^\s*vmov(?:\.w)?\s+(?P<end>\w+),\s*(?P<endf>\w+)",
+                          r"^\s*cmp(?:\.w)?\s+(?P<cnt>\w+),\s*(?P<end>\w+)",
+                               rf"^\s*(cbnz|cbz|bne)(?:\.w)?\s+{lbl}")
+
+    def start(self, loop_cnt, indentation=0, fixup=0, unroll=1, jump_if_empty=None, preamble_code=None, body_code=None, postamble_code=None, register_aliases=None):
+        """Emit starting instruction(s) and jump label for loop"""
+        indent = ' ' * indentation
+        if unroll > 1:
+            assert unroll in [1,2,4,8,16,32]
+            yield f"{indent}lsr {self.additional_data['end']}, {self.additional_data['end']}, #{int(math.log2(unroll))}"
+
+        # Find out by how much the loop counter is modified in one iteration
+        inc_per_iter = 0
+        if body_code is not None:
+            try:
+                loop_cnt_reg = register_aliases[loop_cnt]
+            except KeyError:
+                loop_cnt_reg = loop_cnt
+            for l in body_code:
+                if l.text == "":
+                    continue
+                inst = Instruction.parser(l)
+                if loop_cnt_reg.lower() == inst[0].addr and inst[0].increment is not None:
+                    inc_per_iter = inc_per_iter + simplify(inst[0].increment)
+        logging.debug(f"Loop counter {loop_cnt} is incremented by {inc_per_iter} per iteration")
+        # Check whether instructions modifying the loop count moved to
+        # pre/postamble and adjust the fixup based on that.
+        new_fixup = 0
+        if postamble_code is not None:
+            new_fixup = 0
+            for l in postamble_code:
+                if l.text == "":
+                    continue
+                inst = Instruction.parser(l)
+                if loop_cnt in inst[0].args_in_out and inst[0].increment is not None:
+                    new_fixup = new_fixup + simplify(inst[0].increment)
+
+        if new_fixup != 0 or fixup != 0:
+            yield f"{indent}push {{{self.additional_data['end']}}}"
+            yield f"{indent}vmov {self.additional_data['end']}, {self.additional_data['endf']}"
+
+        if new_fixup != 0:
+            yield f"{indent}sub {self.additional_data['end']}, {self.additional_data['end']}, #{new_fixup}"
```

Comment:
Why don't we merge the instructions for `new_fixup` and `fixup`?

Reply:
this is actually a bug here that has been fixed in the other branch. please update.